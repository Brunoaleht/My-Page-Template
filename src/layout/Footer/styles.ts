import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: solid 0.1rem ${theme.colors.mediumGray};
    display: flex;
    justify-content: center;
    align-items: center;
    ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 1rem;
      padding-bottom: 1rem;
    }
  `}
`;
