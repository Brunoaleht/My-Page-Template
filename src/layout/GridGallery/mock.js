export default {
  background: false,
  title: 'Grid Image',
  description: `Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam officiis quibusdam iste! Nulla, !`,
  grid: [
    {
      altText: 'Text 1',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=1',
    },
    {
      altText: 'Text 2',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=2',
    },
    {
      altText: 'Text 3',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=3',
    },
    {
      altText: 'Text 4',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=4',
    },
    {
      altText: 'Text 5',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=5',
    },
    {
      altText: 'Text 6',
      imgUrl: 'https://source.unsplash.com/random/600x600?r=6',
    },
  ],
};
