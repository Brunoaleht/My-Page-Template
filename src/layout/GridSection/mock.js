export default {
  background: false,
  title: 'Grid Text',
  description: 'description',
  grid: [
    {
      title: 'Text 1',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
    {
      title: 'Text 2',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
    {
      title: 'Text 3',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
  ],
};
