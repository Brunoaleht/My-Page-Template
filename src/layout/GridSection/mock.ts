import { GridSectionProps } from '.';

export default {
  background: false,
  title: 'Grid Text',
  description: 'description',
  grid: [
    {
      id: 1,
      title: 'Text 1',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
    {
      id: 2,
      title: 'Text 2',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
    {
      id: 3,
      title: 'Text 3',
      description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat ipsum officia beatae modi facere perferendis rerum maiores. Ullam quidem culpa commodi a provident perspiciatis deserunt facilis atque dolorum numquam? Iure.`,
    },
  ],
} as GridSectionProps;
