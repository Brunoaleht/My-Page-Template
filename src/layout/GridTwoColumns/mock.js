export default {
  background: false,
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem illo adipisci, optio praesentium veniam omnis ad, quia accusamus qui animi non error commodi. Labore commodi exercitationem perferendis unde minima minus?`,
  imgUrl: 'assets/images/javascript.svg',
};
