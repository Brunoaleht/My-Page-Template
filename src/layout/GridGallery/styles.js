import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-top: 0;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;
export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
