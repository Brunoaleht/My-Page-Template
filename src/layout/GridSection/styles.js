import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../../components/Heading/styles';
import { Container as TextComponent } from '../../components/TextComponent/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const GridItem = styled.div`
  ${() => css`
    ${HeadingContainer} {
      position: relative;
      left: 1rem;
    }
  `}
`;
