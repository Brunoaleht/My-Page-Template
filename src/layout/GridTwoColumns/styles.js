import styled, { css } from 'styled-components';
import { Title } from '../../components/Heading/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;
export const ImgContainer = styled.div`
  ${() => css``}
`;
export const Img = styled.img`
  ${({ theme }) => css`
    position: relative;
    left: 15rem;
    width: 75%;
    @media ${theme.media.lteMedium} {
      left: 0;
    }
  `}
`;
