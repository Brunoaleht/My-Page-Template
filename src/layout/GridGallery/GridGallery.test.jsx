import { renderTheme } from '../../styles/render-theme';
import { GridGallery } from '.';

import mock from './mock';

describe('<GridGallery />', () => {
  it('should render with background', () => {
    const { container } = renderTheme(<GridGallery {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without background', () => {
    const { container } = renderTheme(<GridGallery {...mock} background={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
