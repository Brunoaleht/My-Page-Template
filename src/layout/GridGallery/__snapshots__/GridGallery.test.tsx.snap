// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridGallery /> should render with background 1`] = `
.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-top: 0;
  margin-bottom: 3.2rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <section
        class="c2"
      >
        <h2
          class="c3"
        >
          Grid Image
        </h2>
        <div
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam officiis quibusdam iste! Nulla, !
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Text 1"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=1"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 2"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=2"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 3"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=3"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 4"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=4"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 5"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 6"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=6"
            />
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`<GridGallery /> should render without background 1`] = `
.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-top: 0;
  margin-bottom: 3.2rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <section
        class="c2"
      >
        <h2
          class="c3"
        >
          Grid Image
        </h2>
        <div
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam officiis quibusdam iste! Nulla, !
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Text 1"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=1"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 2"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=2"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 3"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=3"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 4"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=4"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 5"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Text 6"
              class="c8"
              src="https://source.unsplash.com/random/600x600?r=6"
            />
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
